woocommerce/woocommerce-ios
# Edit Product: inventory settings local validation - stock quantity and SKU

Data validation for #1424 

## Changes

- Created `IntegerInputFormatter` for integer input, and used it for the stock quantity input
- Updated the regex formatting in `DecimalInputFormatter` to enforce max 1 leading zero with a test case
- Added a `ProductAction` - `validateProductSKU` - to validate a Product SKU by checking against other existing Products in the storage
- In `ProductInventorySettingsViewController`, when the user saves the inventory settings, dispatched `ProductAction.validateProductSKU` to validate the SKU field. If there is an error, an alert is displayed

## Testing

Prerequisite: at least one Product in the store has a non-empty SKU

- Launch the app
- Go to the Products tab
- Tap on a simple Product that doesn't have a SKU
- Tap "edit"
- Turn on "Manage stock" and play with the stock quantity field --> only integer value should be allowed
- Enter a SKU that is the same as the Product in the prerequisite
- Tap "Done" to save the inventory settings --> there should be an alert saying the SKU is duplicated or invalid (this UI might be replaced by a nicer inline error in the future)
- Edit the SKU to a new value that isn't used by other Products in the store
- Tap "Done" to save the inventory settings --> the SKU should be saved remotely

## Example screenshots

Note: any suggestions for the alert title/message are welcome!

<img src="https://user-images.githubusercontent.com/1945542/71713744-fb2ef080-2e45-11ea-9cc7-79865d39e4b3.png" width="320"/>

Update release notes:
